numpy/lib/tests/test_ufunclike.py

```diff
@@ -53,7 +53,8 @@ def __new__(cls, data, metadata=None):
                 return res
 
             def __array_wrap__(self, obj, context=None):
-                obj.metadata = self.metadata
+                if isinstance(obj, MyArray):
+                    obj.metadata = self.metadata
```

Comment:
This was previously crashing, because `obj` was `ndarray`, and did not have the metadata property.

I think there's a bigger issue here in that `__array_prepare__` is not called either, but with `__array_ufunc__` approaching release, I don't think we need to fix that any more